src/App.tsx

```diff
@@ -23,201 +171,59 @@ export const App: React.FC = () => {
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
-
-            <div data-cy="TodoLoader" className="modal overlay is-active">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+          {filteredTodos.map(todo => (
+            <TodoComponent
+              key={todo.id}
+              todo={todo}
+              getTodo={getTodosFromApi}
+              setErrorMessage={setErrorMessage}
+              editTitle={editTitle}
+              setIsHidden={setIsHidden}
+              isEditting={isEditing}
+              setIsEditting={setIsEditing}
+              setTodoSelected={setTodoSelected}
+            />
+          ))}
+
+          {tempTodo && (
+            <TodoComponent
+              todo={tempTodo}
+              getTodo={getTodosFromApi}
+              setErrorMessage={setErrorMessage}
+              editTitle={editTitle}
+              setIsHidden={setIsHidden}
+              setTodoSelected={setTodoSelected}
+            />
+          )}
         </section>
 
-        <footer className="todoapp__footer" data-cy="Footer">
-          <span className="todo-count" data-cy="todosCounter">
-            4 items left
-          </span>
-
-          <nav className="filter" data-cy="Filter">
-            <a
-              data-cy="FilterLinkAll"
-              href="#/"
-              className="filter__link selected"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="FilterLinkActive"
-              href="#/active"
-              className="filter__link"
-            >
-              Active
-            </a>
-            <a
-              data-cy="FilterLinkCompleted"
-              href="#/completed"
-              className="filter__link"
-            >
-              Completed
-            </a>
-          </nav>
-
-          <button
-            data-cy="ClearCompletedButton"
-            type="button"
-            className="todoapp__clear-completed"
-          >
-            Clear completed
-          </button>
-        </footer>
+        {todos.length > 0 && (
+          <Footer
+            todos={todos}
+            filterType={filterType}
+            setFilterType={setFilterType}
+            clearCompleted={clearCompleted}
+          />
+        )}
       </div>
 
-      <div
-        data-cy="ErrorNotification"
-        className="notification is-danger is-light has-text-weight-normal"
-      >
-        <button
-          data-cy="HideErrorButton"
-          type="button"
-          className="delete"
-        />
-
-        Unable to add a todo
-        <br />
-        Unable to delete a todo
-        <br />
-        Unable to update a todo
-      </div>
+      <ErrorMessage
+        isHidden={isHidden}
```

Comment:
```suggestion
      {errorMessage && <ErrorMessage
```
You don't need `isHidden` at all. It's the same as check `errorMessage === ''`.
Also don't render this component at all if no message.